### ISSUE TITLE
🐛 Hot Reloading crashes app

### ISSUE THREAD

**Issue by MooNag** [googleads/googleads-mobile-flutter#50]
## Bug report

**Describe the bug**
Platform exception if you hot reload an app with this plugin version google_mobile_ads: 0.11.0+2

### Steps to reproduce

1. Implement the plugin following the guide; you do not need to add an ad; but adding a banner ad will cause the app to fully crash without recovery.
2. hot reload the app (Shift + R)
4. See error or incorrect behavior

### Expected behavior

The app hot reloads and displays ads again.

---

## Additional context

Here are some of the errors and logs that help with the issue.

<details><summary>Click To Expand</summary>

```
E/MethodChannel#plugins.flutter.io/google_mobile_ads( 7709): Failed to handle method call
E/MethodChannel#plugins.flutter.io/google_mobile_ads( 7709): java.lang.IllegalArgumentException: Ad for following adId already exists: 0
E/MethodChannel#plugins.flutter.io/google_mobile_ads( 7709):    at io.flutter.plugins.googlemobileads.AdInstanceManager.trackAd(AdInstanceManager.java:66)
E/MethodChannel#plugins.flutter.io/google_mobile_ads( 7709):    at io.flutter.plugins.googlemobileads.GoogleMobileAdsPlugin.onMethodCall(GoogleMobileAdsPlugin.java:244)
E/MethodChannel#plugins.flutter.io/google_mobile_ads( 7709):    at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:233)
E/MethodChannel#plugins.flutter.io/google_mobile_ads( 7709):    at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
E/MethodChannel#plugins.flutter.io/google_mobile_ads( 7709):    at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:818)
E/MethodChannel#plugins.flutter.io/google_mobile_ads( 7709):    at android.os.MessageQueue.nativePollOnce(Native Method)
E/MethodChannel#plugins.flutter.io/google_mobile_ads( 7709):    at android.os.MessageQueue.next(MessageQueue.java:335)
E/MethodChannel#plugins.flutter.io/google_mobile_ads( 7709):    at android.os.Looper.loop(Looper.java:206)
E/MethodChannel#plugins.flutter.io/google_mobile_ads( 7709):    at android.app.ActivityThread.main(ActivityThread.java:8429)
E/MethodChannel#plugins.flutter.io/google_mobile_ads( 7709):    at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#plugins.flutter.io/google_mobile_ads( 7709):    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:596)
E/MethodChannel#plugins.flutter.io/google_mobile_ads( 7709):    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)

E/flutter ( 7709):      at io.flutter.plugins.googlemobileads.AdInstanceManager.trackAd(AdInstanceManager.java:66)
E/flutter ( 7709):      at io.flutter.plugins.googlemobileads.GoogleMobileAdsPlugin.onMethodCall(GoogleMobileAdsPlugin.java:244)
E/flutter ( 7709):      at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:233)
E/flutter ( 7709):      at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
E/flutter ( 7709):      at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:818)
E/flutter ( 7709):      at android.os.MessageQueue.nativePollOnce(Native Method)
E/flutter ( 7709):      at android.os.MessageQueue.next(MessageQueue.java:335)
E/flutter ( 7709):      at android.os.Looper.loop(Looper.java:206)
E/flutter ( 7709):      at android.app.ActivityThread.main(ActivityThread.java:8429)
E/flutter ( 7709):      at java.lang.reflect.Method.invoke(Native Method)
E/flutter ( 7709):      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:596)
E/flutter ( 7709):      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
E/flutter ( 7709): )
E/flutter ( 7709): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:581:7)
E/flutter ( 7709): #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:158:18)
E/flutter ( 7709): <asynchronous suspension>
E/flutter ( 7709): #2      BannerAd.load (package:google_mobile_ads/src/ad_containers.dart:455:5)
E/flutter ( 7709): <asynchronous suspension>

```

</details>

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel beta, 2.0.0, on Microsoft Windows [Version 10.0.19042.804], locale en-US)
[!] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    X Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.
[√] Chrome - develop for the web
[√] Android Studio (version 4.1.0)
[√] VS Code (version 1.53.2)
[√] Connected device (3 available)

```

</details>

**Comment by carteirarica** [comment]
Tried everything, but it keeps crashing the app after hot realoading. 

One could even wraps load method (for bannerad) or initialize (for mobileads.instance) with a try block that it will crash anyway.

**Comment by RR-Reddy** [comment]
Here is error log 
[admob_app_crash_log.txt](https://github.com/googleads/googleads-mobile-flutter/files/6103065/admob_app_crash_log.txt)

![Screenshot 2021-03-08 at 10 53 31 PM](https://user-images.githubusercontent.com/32235964/110357297-253efc80-8061-11eb-8162-9f34250efbe1.png)

**Comment by jjliu15** [comment]
Should be resolved in 0.11.0+4, via #94.
